vassal-app/src/main/java/VASSAL/build/module/StartupGlobalKeyCommand.java

```diff
@@ -139,16 +169,37 @@ public VisibilityCondition getAttributeVisibility(String key) {
     if (List.of(BUTTON_TEXT, TOOLTIP, ICON, HOTKEY).contains(key)) {
       return () -> false;
     }
+    else if (List.of(WHEN_TO_APPLY, HOTKEY_OR_KEY_COMMAND).contains(key)) {
+      return () -> true;
+    }
+    else if (List.of(GLOBAL_HOTKEY).contains(key)) {
+      return () -> SEND_HOTKEY.equals(hotkeyOrKeyCommand);
+    }
     else {
-      return super.getAttributeVisibility(key);
+      //return () -> {
+      //  System.out.println(key + " " + SEND_KEY_COMMAND.equals(hotkeyOrKeyCommand));
+      //  System.out.println(key + " " + super.getAttributeVisibility(key).shouldBeVisible());
+      //  return SEND_KEY_COMMAND.equals(hotkeyOrKeyCommand) && super.getAttributeVisibility(key).shouldBeVisible();
+      //};
+      // THINGS THAT DON'T WORK
+      //return () -> (SEND_KEY_COMMAND.equals(hotkeyOrKeyCommand) && (super.getAttributeVisibility(key)).shouldBeVisible());
+      return new VisibilityCondition.VisibilityAND(() -> SEND_KEY_COMMAND.equals(hotkeyOrKeyCommand), super.getAttributeVisibility(key));
+
+      // THINGS THAT DO "WORK"
+      //return super.getAttributeVisibility(key);
+      //return () -> SEND_KEY_COMMAND.equals(hotkeyOrKeyCommand);
+      //return new VisibilityCondition.VisibilityAND(() -> SEND_KEY_COMMAND.equals(hotkeyOrKeyCommand), () -> SEND_KEY_COMMAND.equals(hotkeyOrKeyCommand));
     }
```

Comment:
We probably don't need all the comments here.

Reply:
lol clean up the body bags